### ISSUE TITLE
Integer encoding not being handled correctly

### ISSUE THREAD

**Issue by jakekgrog** [PeculiarVentures/ASN1.js#88]
Hi there,

_First off, I'm no ASN.1 expert so please correct me and close the issue if I'm wrong._

My understanding is that `INTEGER` types can be positive, negative, zero and have any magnitude.

In most languages you have the ability to specify a size and sign of a number

If the number is signed, then the high-order bit (the first bit of the most significant byte) determines whether the value is negative or positive.

To account for this, I understand ASN.1 Integers are padded with 0x00 if the high order bit is set. Hence a "2048-bit" (256-byte) number may actually use 257 bytes in ASN.1.

In pseudo code that is:

```
# parsing
if 0x00 == bytes[0] {
  bytes = bytes.slice(1)
}
```

```
# packing
if 0x80 & bytes[0] {
  bytes = bytes.prepend(0)
}
```

The prepending of `0x00` doesn't appear to be happening and I've had to manually add a `0x00` byte as the first byte if the first byte is > `0x80`.

As an example:

```javascript
const param = "FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF";

let output = new ASN1.Integer({
    valueHex: new Uint8Array([hexStringToArray(param)]).buffer,
});

console.log(output.toString('hex'))

// The above should output '00FFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF'
// Note the prepended 0x00
// However that isn't happening
```

**Comment by tstiemerling** [comment]
This is correct, and is causing RSA public key decoding to fail in my application. Someone should really fix this.
